### ISSUE TITLE
Distinct / group by support?

### ISSUE THREAD

**Issue by sharq1** [realm/realm-js#474]
It would be very useful to have.

**Comment by ravitez** [comment]
I vote

**Comment by 289420843** [comment]
+1

**Comment by ugendrang** [comment]
+1

**Comment by kevboh** [comment]
also support this idea and am curious if there are workarounds available

**Comment by kristiandupont** [comment]
At the moment the only way to achieve this is to pull all your data and use something like lodash to accomplish it I'm afraid.

**Comment by kevboh** [comment]
Fair enough, thanks for the quick answer. In general, is there any way to see and upvote the realm-js roadmap beyond github issues?

**Comment by kristiandupont** [comment]
Not at the moment, no. However, we do monitor GitHub as it's what we use for our internal project planning so we do take notice when people comment in here.

**Comment by kevboh** [comment]
I see—thanks for clarifying!

**Comment by sajjadrad** [comment]
+1

**Comment by vivekkhurana** [comment]
+1

**Comment by Almjz** [comment]
+1

**Comment by kosiakMD** [comment]
+1

**Comment by faogustavo** [comment]
+1

**Comment by badaz** [comment]
+1

**Comment by andrew-wagner89** [comment]
+1

**Comment by sellmeadog** [comment]
Is this supposed to be supported 2.3.x?  [Per the current documentation page](https://realm.io/docs/javascript/latest#filtering):

_Sorting and find distinct values are possible with functions `SORT` and `DISTINCT`, e.g. age > 20 `SORT(name ASC, age DESC) DISTINCT(name)`._

I have tried to use these functions without success.  I also have [an open question on StackOverflow](https://stackoverflow.com/questions/49638129/realm-react-native-does-sort-distinct-actually-work) that has is still unanswered.

**Comment by reyalpsirc** [comment]
Yes, this seems to work well with a predicate followed by sort and distinct.
I even made it work with a relation: `time != null SORT(activity.id ASC, time DESC) DISTINCT(activity)`
